executor/brie.go

```diff
@@ -0,0 +1,464 @@
+	sc := &stmtctx.StatementContext{TimeZone: b.ctx.GetSessionVars().Location()}
+	t, err := types.ParseTime(sc, ts, mysql.TypeTimestamp, types.MaxFsp)
+	if err != nil {
+		return 0, err
+	}
+	t1, err := t.GoTime(sc.TimeZone)
+	if err != nil {
+		return 0, err
+	}
+	return variable.GoTimeToTS(t1), nil
+}
+
+var brieStmtLabel fmt.Stringer = stringutil.StringerStr("BRIEStmt")
+
+func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema) Executor {
+	e := &BRIEExec{
+		baseExecutor: newBaseExecutor(b.ctx, schema, brieStmtLabel),
+		info: &brieTaskInfo{
+			kind: s.Kind,
+		},
+	}
+
+	tidbCfg := config.GetGlobalConfig()
+	if tidbCfg.Store != "tikv" {
+		b.err = errors.Errorf("%s requires tikv store, not %s", s.Kind, tidbCfg.Store)
+		return nil
+	}
+
+	cfg := task.Config{
+		TLS: task.TLSConfig{
+			CA:   tidbCfg.Security.ClusterSSLCA,
+			Cert: tidbCfg.Security.ClusterSSLCert,
+			Key:  tidbCfg.Security.ClusterSSLKey,
+		},
+		PD:            []string{tidbCfg.Path},
+		Concurrency:   4,
+		Checksum:      true,
+		LogProgress:   true,
+		CaseSensitive: tidbCfg.LowerCaseTableNames == 0,
+		Filter: filter.Rules{
+			DoDBs: s.Schemas,
+		},
+	}
+
+	storageURL, err := url.Parse(s.Storage)
+	if err != nil {
+		b.err = errors.Annotate(err, "invalid destination URL")
+		return nil
+	}
+
+	switch storageURL.Scheme {
+	case "s3":
+		storage.ExtractQueryParameters(storageURL, &cfg.S3)
+	case "gs", "gcs":
+		storage.ExtractQueryParameters(storageURL, &cfg.GCS)
+	default:
+		break
+	}
+
+	cfg.Storage = storageURL.String()
+	e.info.storage = cfg.Storage
+
+	for _, opt := range s.Options {
+		switch opt.Tp {
+		case ast.BRIEOptionRateLimit:
+			cfg.RateLimit = opt.UintValue
+		case ast.BRIEOptionConcurrency:
+			cfg.Concurrency = uint32(opt.UintValue)
+		case ast.BRIEOptionChecksum:
+			cfg.Checksum = opt.UintValue != 0
+		case ast.BRIEOptionSendCreds:
+			cfg.SendCreds = opt.UintValue != 0
+		}
+	}
+
+	if len(s.Tables) != 0 {
+		cfg.Filter.DoTables = make([]*filter.Table, 0, len(s.Tables))
+		for _, tbl := range s.Tables {
+			cfg.Filter.DoTables = append(cfg.Filter.DoTables, &filter.Table{
+				Name:   tbl.Name.O,
+				Schema: tbl.Schema.O,
+			})
+		}
+	}
+
+	switch s.Kind {
+	case ast.BRIEKindBackup:
+		e.backupCfg = &task.BackupConfig{Config: cfg}
+
+		for _, opt := range s.Options {
+			switch opt.Tp {
+			case ast.BRIEOptionLastBackupTS:
+				tso, err := b.parseTSString(opt.StrValue)
+				if err != nil {
+					b.err = err
+					return nil
+				}
+				e.backupCfg.LastBackupTS = tso
+			case ast.BRIEOptionLastBackupTSO:
+				e.backupCfg.LastBackupTS = opt.UintValue
+			case ast.BRIEOptionBackupTimeAgo:
+				e.backupCfg.TimeAgo = time.Duration(opt.UintValue)
+			case ast.BRIEOptionBackupTSO:
+				e.backupCfg.BackupTS = opt.UintValue
+			case ast.BRIEOptionBackupTS:
+				tso, err := b.parseTSString(opt.StrValue)
+				if err != nil {
+					b.err = err
+					return nil
+				}
+				e.backupCfg.BackupTS = tso
```

Comment:
🤔 This can only be triggered by

```sql
backup database * to 's3://bucket/prefix' 
    snapshot = '2020-05-01 00:00:00' 
    snapshot = 1234567890;
```

because of how things are parsed, the last option wins.

I think this is sorta like 

```sql
create table a (a serial) 
    auto_increment = 9999 
    auto_increment = 8888;
```

which doesn't return any errors and sets the auto inc ID to 8888.